### PR TITLE
ci: update Postgres version

### PR DESCRIPTION
**Pull request by eulerto**
Use the latest stable version of PostgreSQL. It covers the latest protocol changes.

### CODE REVIEW

**Comment by JelteF** [comment]
The OSX build seems broken because of this.

**Comment by eulerto** [comment]
It seems the setup errors are gone. Hmm. I don't know why this test is failing like this. @JelteF do you have an idea?

**Comment by JelteF** [comment]
The first error I've seen before when not using the `master` version of `psycopg`. Now that psycopg 3.2 is out though there's no need to use the master version of it anymore. So I think it makes sense to change it to install the released version in `requirements.txt` and then change all the tests that break because we are doing so.

**Comment by eulerto** [comment]
I'm fine with the current PR. I fixed the Mac OS. The alpine is still broken but it is not because of this PR but another issue with TLS. @JelteF objections?

File: .cirrus.yml [hunk 1]
```diff
@@ -195,25 +195,26 @@ task:
...
     - sudo pfctl -f /etc/pf.conf
-  env:
-    CPPFLAGS: -I/opt/homebrew/opt/openssl@3/include
-    LDFLAGS: -L/opt/homebrew/opt/openssl@3/lib
   build_script:
     - ./autogen.sh
     - ./configure --prefix=$HOME/install --enable-werror
     - make -j4
+  env:
+    PATH: /opt/homebrew/opt/postgresql@${PGVERSION}/bin:$PATH
```
Comment (JelteF) [inline .cirrus.yml:215]: can this be merged with the other `env` section?

File: .cirrus.yml [hunk 2]
```diff
@@ -195,25 +195,26 @@ task:
...
   test_script:
-    - source venv/bin/activate && make check -j4 CONCURRENCY=4
+    - source venv/bin/activate && make -j4 check CONCURRENCY=4
```
Comment (JelteF) [inline .cirrus.yml:217]: Was this change actually needed?

**Review by JelteF — APPROVED** [review]
LGTM appart from two small comments. Please note that we should update the required jobs in the github repo settings before (or quickly after) merging.

File: .cirrus.yml (hunk 1 above)
Comment (eulerto) [reply .cirrus.yml:215]: I think so. The main reason I added here was that that path exists only after installing the PostgreSQL package.

File: .cirrus.yml (hunk 2 above)
Comment (eulerto) [reply .cirrus.yml:217]: There is no strong reason. Consistency with the similar commands. I should have mentioned it.

**Comment by eulerto** [comment]
> Please note that we should update the required jobs in the github repo settings before (or quickly after) merging.

What settings are you referring to? AFAICS we are not using GH Actions. Cirrus CI is a third-part app that is integrated into GitHub.

File: .cirrus.yml (hunk 1 above)
Comment (JelteF) [reply .cirrus.yml:215]: Let's merge them then, because it's a bit confusing that we have two `env` keys in the same object. Based on some quick searching it doesn't seem specified what that means according the yaml spec (it seems like github merges the objects)

**Comment by JelteF** [comment]
> What settings are you referring to?

I the "Required" tag after the CI job: 
image

This checks are required for merging into the master branch, but they contain the 14 number. They can be changed in the branch protection settings here: https://github.com/pgbouncer/pgbouncer/settings/branch_protection_rules/27973333

**Comment by JelteF** [comment]
FYI: I made the change I suggested to the `env` entry, merged the PR and updated the branch protection rules
